asmcomp/asmgen.ml

```diff
@@ -156,8 +156,13 @@ let write_ir prefix =
 
 let should_emit () = not (should_stop_after Compiler_pass.Linearization)
 
-let should_use_linscan fun_codegen_options =
-  !use_linscan || List.mem Cmm.Use_linscan_regalloc fun_codegen_options
+let should_use_linscan fd =
+  !use_linscan
+  || List.mem Cmm.Use_linscan_regalloc fd.fun_codegen_options
+  || List.compare_length_with
+       (Reg.all_relocatable_regs ())
```

Comment:
this is a bit tricky:
- if this is called too early, it won't give the right results (e.g., right after reset, before selection)
- current implementation of `Reg.all_relocatable_regs` may be traversing a rather long list, even when using `List.compare_length_with`. It doesn't matter because any register allocator will traverse this list again anyway.

Both of the above are worth a note somewhere at least.

Reply:
I have added a note just before the function,
but it is "safe" in the sense that if called between
reset and selection, the "old" result will be returned.